client/client.py

```diff
@@ -0,0 +1,47 @@
+from pathlib import Path
+
+import requests
+
+from file_data_transfer_api.api_datamodel import FileDatabaseEntry
+
+class Client:
+    def __init__(self):
+        self.url = "http://127.0.0.1:8000"
+
+    def upload_file(self, path_to_file: Path) -> FileDatabaseEntry:
+        file_to_upload = {'file': open(path_to_file, 'rb')}
+        response = requests.post(url=f"{self.url}/files",files=file_to_upload)
```

Comment:
Worth considering this for readability. This clearly splits the root url from the route.

```suggestion
        response = requests.post(
            url=self.url + "/files",
            files=file_to_upload,
        )
```